app/helpers/location_helper.rb

```diff
@@ -134,10 +137,50 @@ def self.filter_by_name(samples_with_metadata, query)
   # See config/initializers/location_name_aliases.rb and add important known
   # aliases there.
   def self.normalize_name_aliases(name, geo_level)
+    # NOTE(jsheu): Provider v1/autocomplete endpoint often has repetitive
+    # country names such as "Cambodia, Cambodia". Dedupe them.
+    if name.include?(", ")
+      parts = name.split(", ")
+      if parts[1] == parts[0]
+        name = parts[0]
+      end
+    end
+
     if LOCATION_NAME_ALIASES.dig(geo_level, name)
       LOCATION_NAME_ALIASES[geo_level][name]
     else
       name
     end
   end
+
+  # Combine results from provider autocomplete and geosearch endpoints for
+  # better results.
+  def self.handle_external_search_results(results)
+    autocomplete_results = results[Location::GEOSEARCH_ACTIONS[0]] || []
+    search_results = results[Location::GEOSEARCH_ACTIONS[1]] || []
+    combined = zip_arrays(autocomplete_results, search_results)
+
+    # - NOTE(jsheu): We get much more relevant results for our use cases from
+    # the 'relation' type, although we don't have a way to solely request those.
+    # OSM relations are used to model logical or geographic relationships
+    # between objects.
+    # - De-dup by name/geo_level and also osm_id.
+    combined
+      .map { |r| adapt_location_iq_response(r) }
+      .select { |r| Location::OSM_SEARCH_TYPES_TO_USE.include?(r[:osm_type]) }
+      .uniq { |r| [r[:name], r[:geo_level]] }
+      .uniq { |r| r[:osm_id] }
+  end
+
+  # Zip/interpolate two arrays. #1 from A, #1 from B, #2 from A, #2 from B...
+  # If one list is longer than the other, extra entries will be appended to the
+  # end of the combined list.
+  def self.zip_arrays(list_a, list_b)
```

Comment:
Thoughts on moving this to `array_helper.rb`? Also renaming it "merge_arrays" since zip already means something?

Reply:
makes sense, extracted 👍